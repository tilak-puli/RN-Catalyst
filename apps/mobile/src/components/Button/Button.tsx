import {View, ViewStyle} from 'react-native';
import {
  Button as RNEButton,
  ButtonProps as RNEButtonProps,
} from '@rneui/themed';

interface ButtonProps extends RNEButtonProps {
  style?: ViewStyle;
}

const Button = ({style = {}, testID, ...rest}: ButtonProps) => {
  const mergedStyles = [style];
  return (
    <View testID={testID} style={mergedStyles}>
      <RNEButton
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...rest}
      />
    </View>
  );
};
export default Button;
