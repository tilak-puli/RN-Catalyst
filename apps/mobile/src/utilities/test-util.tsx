import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {SpinnerProvider} from '../context/spinner/spinnerContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <QueryClientProvider client={query}>
    <SpinnerProvider>{children}</SpinnerProvider>
  </QueryClientProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
