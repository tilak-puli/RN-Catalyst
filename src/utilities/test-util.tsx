import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '@rneui/themed';
import {SpinnerProvider} from '../context/spinner/spinnerContext';
import {GlobalContextProvider} from '../context/global/globalContext';
import theme from '../styles/theme';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <GlobalContextProvider>
    <ThemeProvider theme={theme}>
      <QueryClientProvider client={query}>
        <SpinnerProvider>{children}</SpinnerProvider>
      </QueryClientProvider>
    </ThemeProvider>
  </GlobalContextProvider>
);

export * from '@testing-library/react-native';
export {customRender as render};
