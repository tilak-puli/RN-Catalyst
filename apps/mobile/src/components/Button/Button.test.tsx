import Button from './Button';
import {act, fireEvent, render, screen} from '../../utilities/test-util';

describe('[Component] - [Button]', () => {
  test('should render button component correctly', () => {
    const button = render(<Button title="Press me" onPress={jest.fn()} />);
    expect(button).toMatchSnapshot();
  });

  test('should be clickable', () => {
    const mockFunction = jest.fn();
    render(
      <Button title="Press me" testID="buttonClick" onPress={mockFunction} />,
    );
    act(() => {
      fireEvent.press(screen.getByTestId('buttonClick'));
    });

    expect(mockFunction).toBeCalledTimes(1);
  });
});
