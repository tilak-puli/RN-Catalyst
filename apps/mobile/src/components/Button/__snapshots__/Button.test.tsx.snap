// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - [Button] should render button component correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      Object {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "rippleRadius": undefined,
        "type": "ThemeAttrAndroid",
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#2196F3",
          "borderRadius": 2,
          "elevation": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontWeight": "500",
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      PRESS ME
    </Text>
  </View>
</View>
`;
